### ISSUE TITLE
CrateDB is exposed to ReDoS attack

### ISSUE THREAD

**Issue by mkleen** [crate/crate#11421]
**CrateDB version**:

4.5.1

**Environment description**:

Mac OS X 11.2.3

**Problem description**:

By using a specific regExp query, I am able to cause 100% cpu for a long time.

See https://github.com/opensearch-project/OpenSearch/issues/687

**Steps to reproduce**:

```
create table doc.test(name text) clustered into 1 shards with (number_of_replicas=0);
insert into doc.test values('aaaaaaaaaaaaaaaaaaaaaaaaaasssssssssssssssssssssssssssssss');
refresh table doc.test;
select * from doc.test where name ~ '(.*a){2000}';
```
CPU usage jumps to 100 % for a long time.

**Comment by mkleen** [comment]
https://issues.apache.org/jira/browse/LUCENE-9981

**Comment by mkleen** [comment]
It is already fixed in the Lucene main branch so this will be resolved with the next Lucene release.

**Comment by mkleen** [comment]
This will be fixed with the update to lucene 8.10 https://issues.apache.org/jira/browse/LUCENE-9981
